### ISSUE TITLE
Make it easier to run multiline script

### ISSUE THREAD

**Issue by cowwoc** [pulumi/pulumi-command#558]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details
Please make it easier to run multiline scripts that are declared in a `string` variable. Currently, we need to do something like this:

```
create: "PULUMI_COMMAND_STDOUT=true sh -c '`echo \"${script}\" > ~/script.sh && chmod u+x ~/script.sh && ~/script.sh'"
```

**Comment by mjeffryes** [comment]
Thanks for the suggestion @cowwoc! I can think of a few other workarounds for packaging a multiline script into a single command (bash functions, stuffing '; \' at the end of every line but the last, etc.) but I understand that it would be much nicer if you didn't any packaging like that.

One challenge with just executing a multiline string right now is that Command is actually somewhat agnostic about the shell that it's running in. You could use bash, Tcsh, powershell, or something more obscure, but that means that we probably can't actually just run a multiline script since every shell would probably need a different approach to packaging it as a single command to run. 

What might be doable would be to run a list of commands in order (but as completely separate processes). That would require you to pass a list of strings rather than one multi line string, but would be easier than the cat into the file, then run the file approach you're using. It wouldn't be quite like running a script though, since you could not set variables in one line and then use them later in another.  Another challenge with this approach would be deciding what to do when one of the commands fails; how to represent all the results in state, etc. 

Given all these challenges, I'm not sure this is something we could move on anytime soon, but I'd be very open to more suggestions about how we could make your use case better in light of all the nuance above!

**Comment by mjeffryes** [comment]
I don't think so; I can't run `chmod u+x` because I might not be on a *nix host. You as the user could do that since you know the target: 
1. Put your script in a `StringAsset`([Assets & Archives](https://www.pulumi.com/docs/iac/concepts/assets-archives/) and copy to the host with `CopyToRemote`
2. Use remote.Command to run `chmod u+x script.sh && ./script.sh`

I think the two options for general support that would be most reliable would be:
1. A shorthand for executing multiple independent commands (like I outlined in my first command)
2. Some helper functions that would bundle a multiline string into a single command for a specific shell. eg. `command.createBashMultilineCommand("...")` that would do something equivalent of what you're writing by hand now.
